### PR TITLE
Tokyo Keyboard alix40 initial commit

### PR DESCRIPTION
**Pull request by quadcube**
## Description
1. New keyboard vendor directory under tokyokeyboard
2. ~Moved tokyo60 into new tokyokeyboard directory~
3. Added alix40 with BLE and VIA support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by quadcube** [comment]
@alvn alix40 PR

**Comment by quadcube** [comment]
@fauxpark Hi, could you help me to review this pull request? Thanks!

File: keyboards/tokyokeyboard/alix40/alix40.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include "alix40.h"
```
Comment (zvecr) [inline keyboards/tokyokeyboard/alix40/alix40.c:1]: All alix40 `.c` and `.h` files need GPL compatible headers added

File: keyboards/tokyokeyboard/alix40/alix40.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  matrix_init_user();
+  led_init_ports();
+};
+
+void matrix_scan_kb(void) {
+  // Looping keyboard code goes here
+  // This runs every cycle (a lot)
+  matrix_scan_user();
+};
+
```
Comment (zvecr) [inline keyboards/tokyokeyboard/alix40/alix40.c:15]:
```suggestion

```

File: keyboards/tokyokeyboard/alix40/config.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION ROW2COL
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* Set power consumption to work with mobile devices */
+#define USB_MAX_POWER_CONSUMPTION 500
+
```
Comment (zvecr) [inline keyboards/tokyokeyboard/alix40/config.h:33]:
```suggestion

```

File: keyboards/tokyokeyboard/alix40/readme.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+Hardware Availability: [Tokyo Keyboard](http://tokyokeyboard.com)
+​
+Make example for this keyboard (after setting up your build environment):
+​
+    make tokyokeyboard/alix40:default
+​
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
+
+Note:
+<<<<<<< HEAD
```
Comment (zvecr) [inline keyboards/tokyokeyboard/alix40/readme.md:18]: This merge conflict should be resolved

File: keyboards/tokyokeyboard/alix40/rules.mk [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+AUDIO_ENABLE = no           # Audio output on port C6
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no        # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes      # Enable RGB light
+
+LAYOUTS = 40_alix
+
+EXTRAFLAGS += -flto
```
Comment (zvecr) [inline keyboards/tokyokeyboard/alix40/rules.mk:37]:
```suggestion
```

File: keyboards/tokyokeyboard/alix40/rules.mk [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH = AdafruitBLE    # Enable Bluetooth with the MDBT40 HID (AdafruitBLE)
+BLUETOOTH_ENABLE = OUT_AUTO #
```
Comment (zvecr) [inline keyboards/tokyokeyboard/alix40/rules.mk:27]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
> Moved tokyo60 into new tokyokeyboard directory

This change will have to go via `develop` and not directly to master, I suggest breaking it out into its own PR as to not slow down the rest of the changes.

File: keyboards/tokyokeyboard/alix40/alix40.c (hunk 1 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/alix40.c:15]: Fixed.

File: keyboards/tokyokeyboard/alix40/alix40.c (hunk 2 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/alix40.c]: Amended according to suggested change.

File: keyboards/tokyokeyboard/alix40/config.h (hunk 3 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/config.h]: Amended according to suggested change.

File: keyboards/tokyokeyboard/alix40/readme.md (hunk 4 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/readme.md]: Amended according to suggested change.

File: keyboards/tokyokeyboard/alix40/rules.mk (hunk 5 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/rules.mk]: Amended according to suggested change.

File: keyboards/tokyokeyboard/alix40/rules.mk (hunk 6 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/rules.mk]: Amended according to suggested change.

**Comment by quadcube** [comment]
> > Moved tokyo60 into new tokyokeyboard directory
> 
> This change will have to go via `develop` and not directly to master, I suggest breaking it out into its own PR as to not slow down the rest of the changes.

I have removed tokyo60 folder relocation from this PR. Will open another PR via the `develop` branch. Thank you for the review and suggestions!

File: keyboards/tokyokeyboard/alix40/config.h [hunk 7]
```diff
@@ -0,0 +1,59 @@
+#pragma once
```
Comment (drashna) [inline keyboards/tokyokeyboard/alix40/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/tokyokeyboard/alix40/config.h [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5143 // QC
+#define PRODUCT_ID      0x4134 // A4
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    quadcube
+#define PRODUCT         alix40
+#define DESCRIPTION     q.m.k. keyboard firmware for alix
```
Comment (drashna) [inline keyboards/tokyokeyboard/alix40/config.h:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
Could you add a GPL2+ compatible license header to all of the `c` and `h` files? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/tokyokeyboard/alix40/config.h (hunk 8 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/config.h:25]: Fixed.

File: keyboards/tokyokeyboard/alix40/config.h (hunk 7 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/config.h:15]: Added GPL2+ compatible license header to all .h, .c files. Thanks for the review!

File: keyboards/tokyokeyboard/alix40/readme.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# alix40
+​
+![alix40](https://i2.wp.com/tokyokeyboard.com/wp-content/uploads/2020/08/hello-alix.png)
+​
+Minimal hacker style mechanical keyboard designed in Tokyo. Cherry MX compatible. More Info at [Tokyo Keyboard](http://tokyokeyboard.com).  
+​
+Keyboard Maintainer: [quadcube](https://github.com/quadcube)  
+Hardware Supported: alix40 rev1  
+Hardware Availability: [Tokyo Keyboard](http://tokyokeyboard.com)
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/readme.md:9]:
```suggestion
* Keyboard Maintainer: [quadcube](https://github.com/quadcube)
* Hardware Supported: alix40 rev1
* Hardware Availability: [Tokyo Keyboard](http://tokyokeyboard.com)
```

File: keyboards/tokyokeyboard/alix40/rules.mk [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH = AdafruitBLE    # Enable Bluetooth with the MDBT40 HID (AdafruitBLE)
+AUDIO_ENABLE = no           # Audio output on port C6
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no        # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes      # Enable RGB light
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/rules.mk:32]:
```suggestion

# Processor frequency
F_CPU = 8000000

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
BLUETOOTH = AdafruitBLE
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tokyokeyboard/alix40/keymaps/default/config.h [hunk 11]
```diff
@@ -0,0 +1,17 @@
+/*
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/default/config.h]: config.h removed.

File: keyboards/tokyokeyboard/alix40/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+├─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┤
+│Shift│  Z  │  X  │  C  │  V  │  B  │Enter│  N  │  M  │  ,  │  .  │  /  │
+├─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┤
+│Ctrl │ Alt │     │ Gui │█████│Bksp │Space│█████│ Gui │     │  [  │  ]  │
+└─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┘
+   */
+[0] = LAYOUT_40_alix(
+        KC_TAB,    KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,    KC_P,    KC_GRV,     \
+        KC_ESC,    KC_A,    KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,    KC_L,    KC_SCLN, KC_QUOT,      \
+        KC_LSFT,   KC_Z,    KC_X,   KC_C,   KC_V,   KC_B,   KC_ENT, KC_N,   KC_M,   KC_COMM,   KC_DOT,  KC_SLSH,      \
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/default/keymap.c]: Removed "\".

File: keyboards/tokyokeyboard/alix40/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+├─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┤
+│Shift│  F6 │  F7 │  F8 │  F9 │ F10 │Enter|Next |Vol- |Vol+ | Play│  /  │
+├─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┤
+│Ctrl │ Alt │     │ Gui │█████│Bksp │Space│█████│ Gui │     │  _  │  +  │
+└─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┘
+     */
+[1]= LAYOUT_40_alix(
+        KC_TILD,   KC_EXLM, KC_AT,  KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE,     \
+        KC_DEL,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_SCLN, KC_QUOT,      \
+        KC_LSFT,  KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_ENT,  KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY,  KC_SLSH,      \
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/default/keymap.c]: Removed "\".

File: keyboards/tokyokeyboard/alix40/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+├─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┤
+│Shift│  F6 │  F7 │  F8 │  F9 │ F10 │Enter│Next │PgDn │PgUp │Home │ End │
+├─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┼─────┤
+│Ctrl │ Alt │     │ Gui │█████│Bksp │Space│█████│ Gui │     │  -  │  =  │
+└─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┴─────┘
+ */
+[2]= LAYOUT_40_alix(
+        KC_TILD,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,   KC_7,   KC_8,    KC_9,    KC_0,      KC_BSLS,     \
+        KC_DEL,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, KC_SCLN,   KC_QUOT,      \
+        KC_LSFT,  KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_ENT, KC_MNXT, KC_PGDN, KC_PGUP, KC_HOME,  KC_END,      \
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/default/keymap.c]: Removed " \ ".

File: keyboards/tokyokeyboard/alix40/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+[BL] = LAYOUT_40_alix(
+    KC_TAB,    KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,    KC_P,    KC_GRV,     \
+    KC_ESC,    KC_A,    KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,    KC_L,    KC_SCLN, KC_QUOT,      \
+    KC_LSFT,   KC_Z,    KC_X,   KC_C,   KC_V,   KC_B,   KC_ENT, KC_N,   KC_M,   KC_COMM,   KC_DOT,  KC_SLSH,      \
+    KC_LCTL,   KC_RALT, KC_LGUI,KC_NO,  LT(1, KC_BSPC), LT(2, KC_SPC), KC_NO, KC_RGUI,       KC_LBRC, KC_RBRC),
+
+[FN]= LAYOUT_40_alix(
+    KC_TILD,   KC_EXLM, KC_AT,  KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE,     \
+    KC_DEL,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_SCLN, KC_QUOT,      \
+    KC_LSFT,  KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_ENT,  KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY,  KC_SLSH,      \
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/via/keymap.c]: Removed " \ ".

File: keyboards/tokyokeyboard/alix40/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    KC_LCTL,  KC_RALT, KC_LGUI, KC_NO,  LT(1, KC_BSPC), LT(2, KC_SPC), KC_NO, KC_RGUI,          KC_UNDS, KC_PLUS),
+
+[2]= LAYOUT_40_alix(
+    KC_TILD,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,   KC_7,   KC_8,    KC_9,    KC_0,      KC_BSLS,     \
+    KC_DEL,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, KC_SCLN,   KC_QUOT,      \
+    KC_LSFT,  KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_ENT, KC_MNXT, KC_PGDN, KC_PGUP, KC_HOME,  KC_END,      \
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/via/keymap.c]: Removed " \ ".

File: keyboards/tokyokeyboard/alix40/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+    KC_LCTL,  KC_RALT, KC_LGUI, KC_NO,  LT(1, KC_BSPC), LT(2, KC_SPC), KC_NO, KC_RGUI,         KC_MINS,  KC_EQL),
+
+[3]= LAYOUT_40_alix(
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,     \
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______, _______,   _______,      \
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______, _______,  _______,      \
```
[the remark this answers is not in the document]
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/keymaps/via/keymap.c]: Removed " \ ".

File: keyboards/tokyokeyboard/alix40/readme.md (hunk 9 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/readme.md]: Added " * ".

File: keyboards/tokyokeyboard/alix40/rules.mk (hunk 10 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/rules.mk]: Clean up rules.mk according to your suggested changes. Thanks for reviewing!

File: keyboards/tokyokeyboard/alix40/rules.mk [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+RGBLIGHT_ENABLE = no        # Enable RGB light
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+BLUETOOTH = AdafruitBLE
+
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/rules.mk:32]:
```suggestion
```

File: keyboards/tokyokeyboard/alix40/rules.mk [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/rules.mk:21]:
```suggestion
NKRO_ENABLE = no            # USB Nkey Rollover
```

File: keyboards/tokyokeyboard/alix40/rules.mk [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable RGB light
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/rules.mk:23]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

File: keyboards/tokyokeyboard/alix40/alix40.c [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+void matrix_init_kb(void) {
+  // Keyboard start-up code goes here
+  // Runs once when the firmware starts up
+  matrix_init_user();
+  led_init_ports();
+};
+
+void led_init_ports(void) {
+  /* LEDs on */
+}
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/alix40.c:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tokyokeyboard/alix40/rules.mk (hunk 20 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/rules.mk:23]: Fixed.

File: keyboards/tokyokeyboard/alix40/rules.mk (hunk 19 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/rules.mk]: Fixed.

File: keyboards/tokyokeyboard/alix40/rules.mk (hunk 18 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/rules.mk]: Removed white spaces.

File: keyboards/tokyokeyboard/alix40/alix40.c (hunk 21 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/alix40.c]: Removed redundant code.

File: keyboards/tokyokeyboard/alix40/config.h (hunk 8 above)
Comment (fauxpark) [reply keyboards/tokyokeyboard/alix40/config.h:25]: This still is not fixed.

File: keyboards/tokyokeyboard/alix40/config.h [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 4
+#define MATRIX_COLS 12
+
+/* number of backlight levels */
+#define BACKLIGHT_PIN B7
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 31
+#define BACKLIGHT_ON_STATE 1
```
Comment (fauxpark) [inline keyboards/tokyokeyboard/alix40/config.h:35]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tokyokeyboard/alix40/config.h (hunk 8 above)
Comment (quadcube) [reply keyboards/tokyokeyboard/alix40/config.h]: Ah, my mistake. Now fixed

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**